### ISSUE TITLE
GUI i2p-zero.exe does not open on Windows 10

### ISSUE THREAD

**Issue by snqtbumpng** [knaccc/i2p-zero#33]
Hi.

First time I extracted _i2p-zero-win-gui.v1.20.zip_ for a quick test and ran _i2p-zero.exe_, it opened successfully and also connected. However, after I removed it and re-extracted it days later into another directory, _i2p-zero.exe_ no longer opened. I cannot still get it to open. I run a Windows 10 Enterprise LTSC 10.0.17763 Build 17763 (latest available version).

I do not now what other information I should include since I cannot find any log files and do not know what running processes are possibly related. If there is any, please tell me.

Cheers.

**Comment by knaccc** [comment]
Thanks for the bug report. Please could you go to the ".i2p-zero" folder in your windows home directory, then look in the config/wrapper.log file and copy/paste any errors you see in there

**Comment by snqtbumpng** [comment]
Well, I do not exactly know what constitutes an error. This is part of wrapper.log that, after retractions, I find safe to share--I do not have advanced security knowledge. If any of the lines is significant, I will send the rest of it.
![log](https://user-images.githubusercontent.com/76466709/103024524-3fd6fe80-4565-11eb-8a51-f2ab06987017.jpg)

**Comment by knaccc** [comment]
Thank you - yes, the only sensitive information in that is your username, which you redacted. All I need is that RuntimException - unfortunately the right portion of that exception is cut off so I can't know the exact problem.

Please could you paste the full width, starting with the "java.lang.RuntimeException" line and ending just before the "I2P router will shut down gracefully" line? There will not be anything sensitive in that text.

**Comment by snqtbumpng** [comment]
Does it contain all that is needed?
![log](https://user-images.githubusercontent.com/76466709/103092287-fbeb0480-460b-11eb-84d7-7a579953cb54.jpg)

**Comment by knaccc** [comment]
Hi, that exception is only something that happens during shutdown, so it should not affect anything.

What happens if you try deleting the ".i2p-zero" folder and then running i2p-zero again? If it does not work, please let me know if there is an exception in the wrapper.log file that is different than the one you pasted.

**Comment by snqtbumpng** [comment]
Interesting. _i2p-zero.exe_ DOES open only once every time I delete _.i2p-zero_, and the log file shows the same exception.

**Comment by knaccc** [comment]
Please could you try this:

1. Delete the ".i2p-zero" folder, and run i2p-zero
2. Check wrapper.log to see if there is an exception
3. Stop i2p-zero
4. Check wrapper.log again
5. Start i2p-zero
6. Check wrapper.log again

The idea is to establish exactly the point at which the exception appears. Thanks!

**Comment by snqtbumpng** [comment]
2. None.
4. There it is. Upon stopping, i1p-zero crashes and here is when the log shows the exception.

**Comment by knaccc** [comment]
OK, thanks. The exception is simply because the router is stopping and so the GUI can't access the router to update the bandwidth stats, so is nothing that could be causing the problem.

I don't know why it's not starting a second time, it's quite difficult to diagnose a problem when there is no error to look at. I'll keep trying.

**Comment by knaccc** [comment]
Btw I just tested on Win 10 Enterprise Edition, and it worked for me.

Please let me know if there are any interesting things in the logs in the .i2p-zero/config/logs directory.

**Comment by snqtbumpng** [comment]
Thanks. This is the log file's content:

```
Dec ##, 2020, ##:##:00 #M ERROR [Thread-7    ] net.i2p.router.Router         : Error saving the config to C:\Users\###\.i2p-zero\base\router.config
java.io.IOException: Failed rename from C:\Users\###\.i2p-zero\base\router.config.tmp to C:\Users\###\.i2p-zero\base\router.config
	at combined/net.i2p.data.DataHelper.storeProps(DataHelper.java:560)
	at combined/net.i2p.router.Router.saveConfig(Router.java:1710)
	at combined/net.i2p.router.Router.saveConfig(Router.java:1763)
	at combined/net.i2p.router.Router.warmupCrypto(Router.java:1376)
	at combined/net.i2p.router.Router.startupStuff(Router.java:440)
	at combined/net.i2p.router.Router.runRouter(Router.java:683)
	at org.getmonero.i2p.zero/org.getmonero.i2p.zero.RouterWrapper.lambda$start$1(RouterWrapper.java:94)
	at java.base/java.lang.Thread.run(Thread.java:832)
Dec ##, 2020, ##:##:02 #M WARN  [JobQueue 1/1] uter.startup.LoadClientAppsJob: Warning - No client apps or router console configured - we are just a router
Dec ##, 2020, ##:##:03 #M ERROR [JobQueue 1/1] net.i2p.router.Router         : Error saving the config to C:\Users\###\.i2p-zero\base\router.config
java.io.IOException: Failed rename from C:\Users\###\.i2p-zero\base\router.config.tmp to C:\Users\###\.i2p-zero\base\router.config
	at combined/net.i2p.data.DataHelper.storeProps(DataHelper.java:560)
	at combined/net.i2p.router.Router.saveConfig(Router.java:1710)
	at combined/net.i2p.router.Router.saveConfig(Router.java:1763)
	at combined/net.i2p.router.startup.CreateRouterInfoJob.createRouterInfo(CreateRouterInfoJob.java:163)
	at combined/net.i2p.router.startup.RebuildRouterInfoJob.rebuildRouterInfo(RebuildRouterInfoJob.java:156)
	at combined/net.i2p.router.startup.LoadRouterInfoJob.runJob(LoadRouterInfoJob.java:61)
	at combined/net.i2p.router.JobQueueRunner.runCurrentJob(JobQueueRunner.java:110)
	at combined/net.i2p.router.JobQueueRunner.run(JobQueueRunner.java:66)
Dec ##, 2020, ##:##:03 #M INFO  [JobQueue 4/5] networkdb.reseed.ReseedChecker: Downloading peer router information for a new I2P installation
Dec ##, 2020, ##:##:04 #M INFO  [JobQueue 1/1] ter.transport.udp.UDPTransport: UDP selected random port ###
Dec ##, 2020, ##:##:04 #M ERROR [JobQueue 1/1] net.i2p.router.Router         : Error saving the config to C:\Users\###\.i2p-zero\base\router.config
java.io.IOException: Failed rename from C:\Users\###\.i2p-zero\base\router.config.tmp to C:\Users\###\.i2p-zero\base\router.config
	at combined/net.i2p.data.DataHelper.storeProps(DataHelper.java:560)
	at combined/net.i2p.router.Router.saveConfig(Router.java:1710)
	at combined/net.i2p.router.Router.saveConfig(Router.java:1763)
	at combined/net.i2p.router.transport.ntcp.NTCPTransport.<init>(NTCPTransport.java:293)
	at combined/net.i2p.router.transport.TransportManager.configTransports(TransportManager.java:267)
	at combined/net.i2p.router.transport.TransportManager.startListening(TransportManager.java:455)
	at combined/net.i2p.router.transport.CommSystemFacadeImpl.startup(CommSystemFacadeImpl.java:66)
	at combined/net.i2p.router.startup.BootCommSystemJob.runJob(BootCommSystemJob.java:44)
	at combined/net.i2p.router.JobQueueRunner.runCurrentJob(JobQueueRunner.java:110)
	at combined/net.i2p.router.JobQueueRunner.run(JobQueueRunner.java:66)
Dec ##, 2020, ##:##:06 #M ^^^ 1 similar message omitted ^^^
Dec ##, 2020, ##:##:25 #M WARN  [Reseed      ] uter.networkdb.reseed.Reseeder: EepGet failed on https://reseed.i2p-projekt.de/i2pseeds.su3?netid=2 : java.net.ConnectException: Connection timed out: connect

```

**Comment by knaccc** [comment]
Aha, `Error saving the config to C:\Users\###\.i2p-zero\base\router.config` looks very bad!

Can you think of any permissions issues that might be happening with your system? Are you just logging in normally and running i2p-zero, or are you launching it automatically or as a service in the background?

**Comment by snqtbumpng** [comment]
I see. I sense that the problem does not have anything to do with this implementation but it is with i2p itself, am I correct? Anyway, I am not sure how I can confirm if there is any permission issues. Would you give me a bit of a hint on how to do that? On the second question, I have not ever attempted to run i2p-zero automatically. I only run it manually.

**Comment by knaccc** [comment]
> I see. I sense that the problem does not have anything to do with this implementation but it is with i2p itself, am I correct?

I could be missing something, but yes, I think so. It's the I2P code (net.i2p.data.DataHelper) that is attempting to write the configuration file, and is failing to overwrite the existing file. The error is not specific enough to know why, and so we'd need to alter their code to provide more error detail.

This is the code that is failing: it seems to need to jump through hoops in Windows:

https://github.com/i2p/i2p.i2p/blob/cead0b2fb83dc9fd175452e293f800a409701ca2/core/java/src/net/i2p/util/FileUtil.java#L494

(which is called from here: https://github.com/i2p/i2p.i2p/blob/a0bf76a4b1876e69bfe53b336cf642471fa5b534/core/java/src/net/i2p/data/DataHelper.java#L560)

> I am not sure how I can confirm if there is any permission issues. Would you give me a bit of a hint on how to do that?

Are you just a regular desktop user, that happens to be using Windows Enterprise edition, or is your machine managed as part of some kind of network? Essentially, I'm wondering if some kind of network adminstrator has messed with the way permissions work for your home directory in Windows.

Please could you try right-clicking the .i2p-zero folder that is in your home directory. Then from the right-click menu, choose Properties. Under "Attributes", there will be a box called "Read only (Only applies to files in folder)". It'll either be a box with square black box inside, or an empty black box outline, or a black box outline with a tick inside. Which one of those 3 possibilities are you seeing?

Please also let me know which of those 3 possibilities applies to the Read-only attribute of the "config" directory inside the .i2p-zero folder. Finally, please also let me know the Read-only attribute of the "router.config" file within the config directory.

**Comment by snqtbumpng** [comment]
Thank you.
I am a regular user and use Enterprise LTSC because it is as far as I know more light-weight than the other editions.
The Read-only boxes for both _.i2p-zero_ and _config_ folders have a black square inside. Unfortunately, I could not find _router.config_, as shown in the shot below.
![image](https://user-images.githubusercontent.com/76466709/103454818-d7261980-4cfc-11eb-92ed-cf1d6bf0d690.png)

**Comment by knaccc** [comment]
That's interesting, this means either that it was not possible to create router.config in the first place, or that the overwrite process deleted router.config and could not create a new router.config in its place.

Is there anything of note about the name of your username, i.e. the `###` in `C:\Users\###\.i2p-zero`?

For example, are there any foreign characters, spaces, punctuation, or any characters other than a-z A-Z 0-9 in it?

**Comment by snqtbumpng** [comment]
No. It is my name, a word starting with a upper-case letter followed by a few lower-case letters. All the characters are found in the English alphabet, so no diacritics, etc. It follows the pattern of 'Michael', for example.

Update:
I have just found [this](https://www.reddit.com/r/Monero/comments/kclg5c/for_those_interested_in_running_monerogui_with/). This could be the same as my problem.

**Comment by Gonbatfire** [comment]
Hi, I'm having the same issue, here is my behavior hope it helps:

1st run: Opened i2p zero as normal, didn't touch any settings
2nd run: Opened i2p zero, this time I modified the "max bandwidth" slider 
3rd run: i2p zero won't open anymore
4rd run: I deleted config.json from .i2p-zero\config folder, i2p zero now opens successfully again

And so i2p zero works normal unless I modify the bandwidth slider, even if I move it and then leave it again at the default value, as doing that still creates the config.json, if the file is created the program no longer opens unless I delete it.

I attached the wrapper.log files per each run, I couldn't find anything interesting in there tho sadly:

[wrapper1st run success.log](https://github.com/i2p-zero/i2p-zero/files/9411223/wrapper1st.run.success.log)
[wrapper2nd run success, modified setting.log](https://github.com/i2p-zero/i2p-zero/files/9411224/wrapper2nd.run.success.modified.setting.log)
[wrapper3rd run failure to open.log](https://github.com/i2p-zero/i2p-zero/files/9411225/wrapper3rd.run.failure.to.open.log)
[wrapper4rd run success, after deleting config filey.log](https://github.com/i2p-zero/i2p-zero/files/9411226/wrapper4rd.run.success.after.deleting.config.filey.log)

I'm on Windows 10 pro 21H2 19044.1889
My username consists of my name without any special characters or spaces, starting with a upper-case later, same as "Michael"
Let me know if there is any more info I could provide

**Comment by xialvjun** [comment]
Win11 too.

```
scoop install i2p-zero-gui
// open i2p zero gui and it opens normally
// close it
// try to open it but it doesn't open
// delete C:\Users\###\.i2p-zero ,  my username is just `xialvjun`
// try to open it and it opens
```
